redis-utils/src/main/java/org/ballerinalang/redis/actions/AbstractRedisAction.java

```diff
@@ -1576,20 +1582,33 @@ private BValueArray createBStringArrayFromList(List<String> list) {
         return bMap;
     }
 
-    protected String[] createArrayFromBStringArray(BValueArray bStringArray) {
+    protected static String[] createStringArrayFromBArray(BArray bStringArray) {
         String[] stringArray = new String[(int) bStringArray.size()];
         for (int i = 0; i < bStringArray.size(); i++) {
             stringArray[i] = bStringArray.getString(i);
         }
         return stringArray;
     }
 
-    protected Map<String, String> createMapFromBMap(BMap<String, BString> bMap) {
-        Map<String, String> map = new HashMap<>();
-        bMap.getMap().keySet().forEach(item -> map.put(item, bMap.get(item).stringValue()));
+    protected static Map<String, Object> createMapFromBMap(BMap<String, BString> bMap) {
+        Map<String, Object> map = new HashMap<>();
+        String[] keys = bMap.getKeys();
+        for (int i = 0; i < keys.length; i++) {
+            map.put(keys[i], bMap.get(keys[i]));
+        }
         return map;
     }
 
+    protected static BArray createBstringArrayFromBMap(BMap<String, BString> bMap) {
+        String[] keys = bMap.getKeys();
+        BArray bStringArray = new ArrayValueImpl(new String[bMap.size()]);
+        for (int i = 0; i < keys.length; i++) {
+            bStringArray.add(i, bMap.get(keys[i]).stringValue());
+            i++;
+        }
+        return bStringArray;
+    }
+
     protected void setNullableReturnValues(BValue result, Context context) {
```

Comment:
What is the usage of this method?

Reply:
This method is no longer used. Will remove.

Reply:
Updated in e26d6d2